go/store/prolly/tree/map.go

```diff
@@ -42,7 +42,8 @@ func DiffOrderedTrees[K, V ~[]byte, O Ordering[K]](
 	from, to StaticMap[K, V, O],
 	cb DiffFn,
 ) error {
-	differ, err := DifferFromRoots[K](ctx, from.NodeStore, to.NodeStore, from.Root, to.Root, from.Order)
+	// TODO: If the schema has changed, do we want every row to be considered changed?
```

Comment:
The only interesting case is when a new, nullable column is added to the end of a table, right? That's the one case where we don't rewrite the table for the significant schema change as far as I know. Other less significant schema changes that wouldn't require a full table rewrite would be things like changing the length of a `varchar` field. 

Sounds like you're saying you think we may have this same issue with diff generation for a table's history?